plugins/a11y-keyboard/msd-a11y-keyboard-atspi.c

```diff
@@ -128,7 +128,7 @@ msd_a11y_keyboard_atspi_start (MsdA11yKeyboardAtspi *self)
         /* init AT-SPI if needed */
         atspi_init ();
 
-        self->listener = atspi_device_listener_new (on_key_press_event,
+        self->listener = atspi_device_listener_new ((AtspiDeviceListenerCB)on_key_press_event,
```

Comment:
That's not a very reasonable fix, it's only hiding the warning.

However, there's a couple of things to consider here:
* clang is silly, the *only* issue in the signature is that the function pointer has a `const` first argument, whereas the function expects a function pointer with non-`const` first argument -- e.g. the passed-in function pointer is more restrictive, not less.  I'm not 110% sure how to read the standard, but this seems wrong (`int a = 42; const int *b = &a;` is perfectly valid).  But meh, I guess they know better what they are doing, as [this](https://discourse.llvm.org/t/clang-16-notice-of-potentially-breaking-changes/65562) suggests it's perfectly intentional -- and I can't find a better documentation.
* The libatspi2 docs are confusing, because [they mark the argument as `const`](https://docs.gtk.org/atspi2/callback.DeviceListenerCB.html) yet the [code disagrees](https://gitlab.gnome.org/GNOME/at-spi2-core/-/blob/main/atspi/atspi-device-listener.h#L48).  Actually it apparently [changed](https://gitlab.gnome.org/GNOME/at-spi2-core/-/commit/7dfb0b7fc2d1710ef7fad54f910fa4c6a5e3af17) "recently".

Anyhow, IMO this isn't a proper fix, and would hide an issue if that changed.

I'll prepare a more complete fix, with the leak removed.